apiserver/facades/client/modelmanager/modelmanager_test.go

```diff
@@ -1101,6 +1224,63 @@ func (s *modelManagerStateSuite) setAPIUser(c *gc.C, user names.UserTag) {
 	s.modelmanager = modelmanager
 }
 
+// expectCreateModelStateSuite expects all the calls to the services made during
+// model creation. Since this is the state suite, we are not explicitly
+// testing the services calls here so these expectations are quite permissive.
+func (s *modelManagerStateSuite) expectCreateModelStateSuite(
+	c *gc.C,
+	ctrl *gomock.Controller,
+	modelCreateArgs params.ModelCreateArgs,
+) {
+	modelUUID := modeltesting.GenModelUUID(c)
+	userTag, err := names.ParseUserTag(modelCreateArgs.OwnerTag)
+	c.Assert(err, gc.IsNil)
+	ownerName := user.NameFromTag(userTag)
+	ownerUUID := usertesting.GenUserUUID(c)
+
+	// Get the default cloud name and credential.
+	s.modelService.EXPECT().DefaultModelCloudNameAndCredential(
+		gomock.Any()).Return("dummy", credential.Key{}, nil)
+	// Get the uuid of the model owner.
+	s.accessService.EXPECT().GetUserByName(
+		gomock.Any(), ownerName,
+	).Return(user.User{UUID: ownerUUID}, nil)
+
+	// Create model in controller database.
+	s.modelService.EXPECT().CreateModel(gomock.Any(), model.ModelCreationArgs{
+		Name:        modelCreateArgs.Name,
+		Owner:       ownerUUID,
+		Cloud:       "dummy",
+		CloudRegion: "dummy-region",
+		Credential:  credential.Key{},
+	}).Return(
+		modelUUID,
+		func(context.Context) error { return nil },
+		nil,
+	)
+
+	// Expect call to get the model service factory.
+	modelServiceFactory := mocks.NewMockModelServiceFactory(ctrl)
+	s.serviceFactoryGetter.EXPECT().ServiceFactoryForModel(gomock.Any()).Return(modelServiceFactory)
+
+	// Expect calls to get various model services.
+	modelInfoService := mocks.NewMockModelInfoService(ctrl)
```

Comment:
This all looks copy pasted from the other function no?

Reply:
Pretty much, but they are more permissive because this is the state suite and we're not actually testing this stuff here